---
title: "2.AlcoholConsumptionADBellenguez"
author: "Roopal Desai"
date: "26/04/2022"
output: pdf_document
---
```{r}
rm(list=ls())
```

```{r echo = FALSE, message=FALSE, warning=FALSE}
library("plyr") 
library("tidyverse")
library("stringr") 
library("vroom") 
library("data.table") 
library("MendelianRandomization") 
library("ieugwasr")
```

# 1. Read in the data for Alcohol Consumption and create the genetic instrument at genome wide signifcance level 

```{r}

#Data=for Alcohol Consumptoion from Evangelou et al., 2019 
#MarkerName – CHR:POS:TYPE where POS is NCBI build 37/hg19 and TYPE is SNP or INDEL
#Allele1 – the Effect Allele
#Allele2 – the other Allele
#Freq1 – frequency of effect allele A1
#Effect – effect of Allele 1, on the untransformed mmHg scale
#StdErr – SE of Beta
#P – p-value
#direction - direction of the effect in UKB and AlcGen + CHARGE+ meta-analysis
#TotalSampleSize – total analysed N (i.e. max ~750k)

 AC.data <- vroom("/Volumes/One Touch/Papers /Submitted/4. RiskfactorsMRpaper /Data/Alcohol Consumption/Evangelou et al., 2019 /EvangelouE_31358974.txt")
head(AC.data)

```

```{r}

AC.data = AC.data %>% select("markername","allele1","allele2","effect","stderr","pvalue")
AC.data = AC.data %>% dplyr::rename(Alcohol_a1 = allele1, Alcohol_a2 = allele2, Alcohol_beta = effect, Alcohol_se = stderr, Alcohol_pval = pvalue)

dim(AC.data)
```



```{r}
AC.iv = AC.data[which(AC.data$Alcohol_pval <5*10^-8),]
dim(AC.iv)
head(AC.iv)
```

```{r}
#read in the variant file which contains the rsids and base position and Chromosome for UKBB and is built on GRCH37  
annot <- vroom("/Volumes/One Touch/Papers /Submitted/4. RiskfactorsMRpaper /Data/Blood Pressure/UKBioBankData/variants.tsv.bgz")
head(annot)
annot= annot %>% select("chr", "pos", "rsid")
```


```{r}
AC.iv<-AC.iv %>% separate(markername, c("chr", "pos"), ":")
head(AC.iv)
```
```{r}
ACmerged.iv =merge(x=annot, y=AC.iv, by=c("chr", "pos"))
```



#2. Read in the data for Alzheimer's disease 

```{r}
#AD data from Bellenguez et al., 2022 
#Chromosome and position based on build GRCh38
AD.data <- vroom("/Volumes/One Touch/Papers /Submitted/4. RiskfactorsMRpaper /Data/Alzheimer's Disease/Bellenguez 2022/GCST90027158_buildGRCh38.tsv")
head(AD.data)
```



```{r}
AD.data = AD.data %>% select("variant_id","effect_allele","other_allele","beta","standard_error","p_value")
AD.data = AD.data %>% dplyr::rename(AD_effectallele = effect_allele, AD_noneffectallele = other_allele, AD_beta_not_aligned = beta, AD_se = standard_error, AD_pval = p_value)
```

# 3. Merge the two data sets 

```{r}

colnames(AD.data)[1]=("rsid")
data.merge =merge(ACmerged.iv, AD.data, by="rsid") 

dim(data.merge)
```

# 4. Harmonise and Prune the merged data 

```{r}

# Create a TRUE/FALSE vector to indicate if Alcohol_a1 is equal to AD_effectallele. 

table(toupper(data.merge$Alcohol_a1) == data.merge$AD_effectallele)
table(toupper(data.merge$Alcohol_a1) == data.merge$AD_noneffectallele)
```


```{r}

# The harmonisation of the effect alleles can be performed as below in this case we are just creating the variable AD_beta as all the alleles are aligned 

data.merge$AD_beta = ifelse(toupper(data.merge$Alcohol_a1) == data.merge$AD_effectallele, 
                                data.merge$AD_beta_not_aligned, -1*data.merge$AD_beta_not_aligned)
```

```{r}

#Clump and prune

#The final step is to prune or clump the SNPs. Pruning removes SNPs which are correlated 
# (measured by the squared correlation r2). From a group of correlated SNPs it retains the one with the 
# lowest $p$-value for the exposure. Use the function ieugwasr::ld_clump to prune the data. 
#The algorithm needs to know the rs identifier of the genetic variants (labeled as rsid) and 
#the $p$-value of the risk factor or exposure (labeled as pval). Rename the following columns accordingly:

colnames(data.merge)[1]="rsid"
colnames(data.merge)[8]="pval"


data.clump = ieugwasr::ld_clump(data.merge)
dim(data.clump)
```

```{r}
#Calculate instrument strength
F=mean((data.merge$Alcohol_beta/data.merge$Alcohol_se)^2)
F
```

# 5. Run the Mendelian Randomization 


```{r}
rs = data.clump$rsid
Alcohol_beta = data.clump$Alcohol_beta
Alcohol_se = data.clump$Alcohol_se
AD_beta = data.clump$AD_beta
AD_se = data.clump$AD_se

mr.input = mr_input(bx = Alcohol_beta, bxse = Alcohol_se, by = AD_beta, byse = AD_se, 
                    exposure = "Alcohol Consumption", outcome = "Alzheimer's Disease", snps = rs)
mr_ivw(mr.input)
mr_plot(mr.input, interactive=FALSE)

```
```{r}
mr_allmethods(mr.input)
mr_plot(mr_allmethods(mr.input, method = "main"))
```



