---
title: "5.EducationADBellenguez"
author: "Roopal Desai"
date: "25/04/2022"
output: pdf_document
---

```{r}
rm(list=ls())
```


```{r echo = FALSE, message=FALSE, warning=FALSE}
library("plyr") 
library("tidyverse")
library("stringr") 
library("vroom") 
library("data.table") 
library("MendelianRandomization") 
library("ieugwasr")
```

# 1. Read in the data for Educational Attainment and create the genetic instrument at genome wide signifcance level 

```{r}
#Data= from Okbay et al., 2016 
#Educational attainment is measured in all main analyses as the number 
#of years of schooling completed (EduYears, n=293,723, mean=14.3, #standard deviation=3.6) 

Education.data <- vroom("/Volumes/One Touch/Papers /11. RiskfactorsMRpaper /Data/Educational Attainment/Education Okbay et al., 2016/Okbay_27225129-EduYears_Main.txt")
dim(Education.data)
```

```{r}
head(Education.data)
```

```{r}
Education.data = Education.data %>% select("MarkerName","A1","A2","Beta","SE","Pval")
Education.data = Education.data %>% dplyr::rename(Education_a1 = A1, Education_a2 = A2, Education_beta = Beta, Education_se = SE, Education_pval = Pval)

dim(Education.data)
```


```{r}
#Standardise the beta and se by dividing by the standard deviation 

Education.data$Education_beta<-  Education.data$Education_beta/3.6
Education.data$Education_se <- Education.data$Education_se/3.6
```


```{r}
Education.iv = Education.data[which(Education.data$Education_pval <5*10^-8),]
```

#2. Read in the data for Alzheimer's disease 

```{r}
#AD data from Bellenguez et al., 2022 
#Chromosome and position based on build GRCh38
AD.data <- vroom("/Volumes/One Touch/Papers /11. RiskfactorsMRpaper /Data/Alzheimer's Disease/Bellenguez 2022/GCST90027158_buildGRCh38.tsv")
head(AD.data)
```

```{r}

dim(AD.data)
head(AD.data)
```

```{r}
AD.data = AD.data %>% select("variant_id","effect_allele","other_allele","beta","standard_error","p_value")
AD.data = AD.data %>% dplyr::rename(AD_effectallele = effect_allele, AD_noneffectallele = other_allele, AD_beta_not_aligned = beta, AD_se = standard_error, AD_pval = p_value)
```

# 3. Merge the two data sets 

```{r}

colnames(AD.data)[1]=("MarkerName")
data.merge =merge(Education.iv, AD.data, by="MarkerName") 

dim(data.merge)
```

# 4. Harmonise and Prune the merged data 

```{r}

# Create a TRUE/FALSE vector to indicate if Education_a1 is equal to AD_effectallele. 

table((data.merge$Education_a1) == data.merge$AD_effectallele)
table((data.merge$Education_a1) == data.merge$AD_noneffectallele)
```

```{r}

# The harmonisation of the effect alleles can be performed as below in this case we are just creating the variable AD_beta as all the alleles are aligned 

data.merge$AD_beta = ifelse((data.merge$Education_a1) == data.merge$AD_effectallele, 
                                data.merge$AD_beta_not_aligned, -1*data.merge$AD_beta_not_aligned)
```

```{r}

#Clump and prune

#The final step is to prune or clump the SNPs. Pruning removes SNPs which are correlated 
# (measured by the squared correlation r2). From a group of correlated SNPs it retains the one with the 
# lowest $p$-value for the exposure. Use the function ieugwasr::ld_clump to prune the data. 
#The algorithm needs to know the rs identifier of the genetic variants (labeled as rsid) and 
#the $p$-value of the risk factor or exposure (labeled as pval). Rename the following columns accordingly:

colnames(data.merge)[1]="rsid"
colnames(data.merge)[6]="pval"


data.clump = ieugwasr::ld_clump(data.merge)
dim(data.clump)
```

```{r}
#Calculate instrument strength
F=mean((data.merge$Education_beta/data.merge$Education_se)^2)
F
```

# 5. Run the Mendelian Randomization 


```{r}
rs = data.clump$rsid
Education_beta = data.clump$Education_beta
Education_se = data.clump$Education_se
AD_beta = data.clump$AD_beta
AD_se = data.clump$AD_se

mr.input = mr_input(bx = Education_beta, bxse = Education_se, by = AD_beta, byse = AD_se, 
                    exposure = "Educational Attainment", outcome = "Alzheimer's Disease", snps = rs)
mr_ivw(mr.input)
mr_plot(mr.input, interactive=FALSE)

```

```{r}
mr_allmethods(mr.input)
mr_plot(mr_allmethods(mr.input, method = "main"))
```
