---
title: "6.HearingLossADBellenguez"
author: "Roopal Desai"
date: "26/04/2022"
output: pdf_document
---
```{r}
rm(list=ls())
```

```{r echo = FALSE, message=FALSE, warning=FALSE}
library("plyr") 
library("tidyverse")
library("stringr") 
library("vroom") 
library("data.table") 
library("MendelianRandomization") 
library("ieugwasr")
```

# 1. Read in the data for ARHI and create the genetic instrument at genome wide signifcance level 

```{r}
#Data= from Kalra et al., 2020  
#Age-related hearing impairment (ARHI)
#Based on build 37

ARHI <- vroom("/Volumes/One Touch/Papers /11. RiskfactorsMRpaper /Data/Hearing Loss/Kalra et al.,2020 /GCST90012115_buildGRCh37.tsv")
dim(ARHI)
```


```{r}
head(ARHI)
```


```{r}
ARHI = ARHI %>% select("variant_id","a1","a2","beta","standard_error","p_value")
ARHI = ARHI %>% dplyr::rename(ARHI_a1 = a1, ARHI_a2 = a2, ARHI_beta = beta, ARHI_se = standard_error, ARHI_pval = p_value)

dim(ARHI)
```

```{r}
ARHI.iv = ARHI[which(ARHI$ARHI_pval <5*10^-8),]
dim(ARHI.iv)
```

```{r}
head(ARHI.iv)
```


#2. Read in the data for Alzheimer's disease 

```{r}
#AD data from Bellenguez et al., 2022 
#Chromosome and position based on build GRCh38
AD.data <- vroom("/Volumes/One Touch/Papers /11. RiskfactorsMRpaper /Data/Alzheimer's Disease/Bellenguez 2022/GCST90027158_buildGRCh38.tsv")
head(AD.data)
```

```{r}
dim(AD.data)
```

```{r}
AD.data = AD.data %>% select ("variant_id","effect_allele","other_allele","beta","standard_error","p_value")
AD.data = AD.data %>% dplyr::rename(rsid= variant_id, AD_effectallele = effect_allele, AD_noneffectallele = other_allele, AD_beta_not_aligned = beta, AD_se = standard_error, AD_pval = p_value)
```


# 3. Merge the two data sets 


```{r}
colnames(AD.data)[1]=("variant_id")
data.merge =merge(ARHI.iv, AD.data, by="variant_id") 
head(data.merge)
```


```{r}
dim(data.merge)
```
# 4. Harmonise and Prune the merged data 

```{r}

# Create a TRUE/FALSE vector to indicate if ARHI_a1 is equal to AD_effectallele. 

table((data.merge$ARHI_a1) == data.merge$AD_effectallele)
table((data.merge$ARHI_a1) == data.merge$AD_noneffectallele)
```


```{r}

inconsistent=which((data.merge$ARHI_a1) != data.merge$AD_effectallele 
                   & (data.merge$ARHI_a1) != data.merge$AD_noneffectallele)
data.merge[inconsistent,]
data.merge = data.merge[-inconsistent,]
dim(data.merge)
```

```{r}

# The harmonisation of the effect alleles can be performed as follows: 

data.merge$AD_beta = ifelse((data.merge$ARHI_a1) == data.merge$AD_effectallele,data.merge$AD_beta_not_aligned, -1*data.merge$AD_beta_not_aligned)
```


```{r}
#Clump and prune
#The algorithm needs to know the rs identifier of the genetic variants (labeled as rsid) and 
#the $p$-value of the risk factor or exposure (labeled as pval). Rename the following columns accordingly:

colnames(data.merge)[1]="rsid"
colnames(data.merge)[6]="pval"

data.clump = ieugwasr::ld_clump(data.merge)
dim(data.clump)
```
```{r}
#Calculate instrument strength
F=mean((data.merge$ARHI_beta/data.merge$ARHI_se)^2)
F
```

# 5. Run the Mendelian Randomization 
```{r}
rs = data.clump$rsid
ARHI_beta = data.clump$ARHI_beta
ARHI_se = data.clump$ARHI_se
AD_beta = data.clump$AD_beta
AD_se = data.clump$AD_se

mr.input = mr_input(bx = ARHI_beta, bxse = ARHI_se, by = AD_beta, byse = AD_se, 
                    exposure = "Age Related Hearing Impairment", outcome = "Alzheimer's Disease", snps = rs)
mr_ivw(mr.input)
mr_plot(mr.input, interactive=FALSE)

```


```{r}
mr_allmethods(mr.input)
mr_plot(mr_allmethods(mr.input, method = "main"))
```
