---
title: "Smoking-AD-Bellenguez2022"
author: "Roopal Desai"
date: "11/04/2022"
output: pdf_document
---

```{r}
rm(list=ls())
```


```{r echo = FALSE, message=FALSE, warning=FALSE}
library("plyr") 
library("tidyverse")
library("stringr") 
library("vroom") 
library("data.table") 
library("MendelianRandomization") 
library("ieugwasr")
```

# 1. Read in the data for Lifetime Smoking and create the genetic instrument at genome wide signifcance level 

```{r}
#Data = from Wootton et al., 2019 on lifetime smoking
#SNP			rs number for each single nucelotide polymorphism
#CHR			chromosome
#BP			base position
#EFFECT_ALLELE	allele for which the effect size corresponds
#OTHER_ALLELE	allele for which the effect size does not correspond
#EAF			effect allele frequency
#INFO			imputation quality
#BETA			effect size from BOLT-LMM approximation to infinitesimal mixed model (unstandardised)
#SE			standard error of effect size
#P			infinitesimal mixed model association test p-value
Smoking <- vroom("/Volumes/One Touch/Papers /11. RiskfactorsMRpaper /Data/Smoking/Lifetime Smoking/2019.10.02 Lifetime Smoking GWAS Data Sheet 1.txt")
dim(Smoking)
```

````{r}
head(Smoking)
```

```{r}
Smoking = Smoking %>% select("SNP","EFFECT_ALLELE","OTHER_ALLELE","BETA","SE","P")
Smoking = Smoking %>% dplyr::rename(Smoking_a1 = EFFECT_ALLELE, Smoking_a2 = OTHER_ALLELE, Smoking_beta = BETA, Smoking_se = SE, Smoking_pval = P)


dim(Smoking)
```

```{r}
#Standardise the beta by dividing by the standard deviation SD= 0.6940093 (this is reported in the Readme file)

Smoking$Smoking_beta <- Smoking$Smoking_beta/0.6940093
Smoking$Smoking_se <-Smoking$Smoking_se/0.6940093
```

```{r}
Smoking.iv = Smoking[which(Smoking$Smoking_pval <5*10^-8),]
dim(Smoking.iv)
```
#2. Read in the data for Alzheimer's disease 

```{r}
#AD data from Bellenguez et al., 2022 
#Chromosome and position based on build GRCh38
AD.data <- vroom("/Volumes/One Touch/Papers /11. RiskfactorsMRpaper /Data/Alzheimer's Disease/Bellenguez 2022/GCST90027158_buildGRCh38.tsv")
head(AD.data)
```

```{r}
dim(AD.data)
```

```{r}
AD.data = AD.data %>% select("variant_id","effect_allele","other_allele","beta","standard_error","p_value")
AD.data = AD.data %>% dplyr::rename(SNP=variant_id, AD_effectallele = effect_allele, AD_noneffectallele = other_allele, AD_beta_not_aligned = beta, AD_se = standard_error, AD_pval = p_value)
```

# 3. Merge the two data sets 
`
```{r}
data.merge =merge(Smoking.iv, AD.data, by="SNP") 

dim(data.merge)
```

# 4. Harmonise and Prune the merged data 

```{r}

# Create a TRUE/FALSE vector to indicate if Smoking_a1 is equal to AD_effectallele. 
table((data.merge$Smoking_a1) == data.merge$AD_effectallele)
table((data.merge$Smoking_a1) == data.merge$AD_noneffectallele)
```

```{r}
# The harmonisation of the effect alleles can be performed as follows: 

data.merge$AD_beta = ifelse((data.merge$Smoking_a1) == data.merge$AD_effectallele,                                        data.merge$AD_beta_not_aligned,-1*data.merge$AD_beta_not_aligned)
```

```{r}

#Clump and prune

#The final step is to prune or clump the SNPs. Pruning removes SNPs which are correlated 
# (measured by the squared correlation r2). From a group of correlated SNPs it retains the one with the 
# lowest $p$-value for the exposure. Use the function ieugwasr::ld_clump to prune the data. 
#The algorithm needs to know the rs identifier of the genetic variants (labeled as rsid) and 
#the $p$-value of the risk factor or exposure (labeled as pval). Rename the following columns accordingly:

colnames(data.merge)[1]="rsid"
colnames(data.merge)[6]="pval"


data.clump = ieugwasr::ld_clump(data.merge)
dim(data.clump)
```


```{r}
#Calculate instrument strength
F=mean((data.merge$Smoking_beta/data.merge$Smoking_se)^2)
F
```
# 5. Run the Mendelian Randomization 


```{r}
rs = data.clump$rsid
Smoking_beta = data.clump$Smoking_beta
Smoking_se = data.clump$Smoking_se
AD_beta = data.clump$AD_beta
AD_se = data.clump$AD_se

mr.input = mr_input(bx = Smoking_beta, bxse = Smoking_se, by = AD_beta, byse = AD_se, 
                    exposure = "Smoking", outcome = "Alzheimer's Disease", snps = rs)
mr_ivw(mr.input)
mr_plot(mr.input, interactive=FALSE)

```

```{r}
MRALL <- mr_allmethods(mr.input)
MRALL
mr_plot(mr_allmethods(mr.input, method = "main"))
```


